pybamm/solvers/base_solver.py

```diff
@@ -481,6 +507,11 @@ def solve(self, model, t_eval, external_variables=None, inputs=None):
         if len(model.rhs) == 0 and len(model.algebraic) == 0:
             raise pybamm.ModelError("Cannot solve empty model")
 
+        # t_eval can only be None if the solver is an algebraic solver. In that case
+        # set it to 0
+        if self.algebraic_solver is True and t_eval is None:
+            t_eval = np.array([0])
+
```

Comment:
should we need to raise an error here if t_eval is None and the solver isn't algebraic?